### PR TITLE
chore(tp4): Update sample extension for TP4

### PR DESCRIPTION
**Pull request by dsimansk**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -127,22 +92,15 @@
       </plugin>
 
       <plugin>
-        <groupId>io.syndesis</groupId>
-        <artifactId>syndesis-maven-plugin</artifactId>
-        <version>${syndesis.version}</version>
+        <groupId>io.syndesis.extension</groupId>
+        <artifactId>extension-maven-plugin</artifactId>
```
Comment (zregvart) [inline pom.xml]: I think we need to specify the version here, as far as I understand, the plugin version is not imported from the BOM.

**Comment by dsimansk** [comment]
@zregvart updated, thanks.

File: src/main/resources/META-INF/syndesis/syndesis-extension-definition.json [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
     "id": "damage-report",
     "name": "Damage Reporter",
     "description": "Generates contact list for damaged items",
     "descriptor": {
       "kind": "ENDPOINT",
       "entrypoint": "direct:damage-report",
+      "resource": "classpath:META-INF/syndesis/extensions/extension.xml",
       "inputDataShape": {
         "kind": "java",
         "type": "java.lang.String",
```
Comment (lburgazzoli) [inline src/main/resources/META-INF/syndesis/syndesis-extension-definition.json]: I think `specification` can be omitted and leave the plugin to generate it, ca you give it a try ?

Comment (dsimansk) [reply src/main/resources/META-INF/syndesis/syndesis-extension-definition.json]: @lburgazzoli yes extension is generating dataShapes properly. I've pushed an update.

**Review by zregvart — APPROVED** [review]
I think we're good to go here. Thanks :+1:

**Pull request merged**